web/src/search.js

```diff
@@ -111,11 +133,30 @@ export function initialize({on_narrow_search}) {
                 $search_query_box.trigger("blur");
             }
         });
+
+    // We don't want to make this a focus handler because selecting the
+    // typehead seems to trigger this (and we don't want to open search
+    // when an option is selected and we're closing search).
+    // Instead we explicitly initiate search on click and on specific keyboard
+    // shortcuts.
+    $search_query_box.on("click", initiate_search);
+
+    $(".search_icon").on("mousedown", (e) => {
+        e.preventDefault();
+        // Clicking on the collapsed search box's icon opens search, but
+        // clicking on the expanded search box's search icon does nothing.
+        if ($(e.target).parents(".navbar-search.expanded").length === 0) {
+            initiate_search();
+        }
+    });
 }
 
 export function initiate_search() {
-    message_view_header.open_search_bar_and_close_narrow_description();
-    $("#search_query").typeahead("lookup").trigger("select");
+    // Don't reset the search bar if it's already open.
+    if (!$(".navbar-search.expanded").length) {
```

Comment:
This is the Boolean example with `.navbar-search.expanded`

Reply:
Switched to `if ($(".navbar-search.expanded").length > 0)` :)